src/sentry-client/src/FbwAircraftSentryClient.ts

```diff
@@ -0,0 +1,189 @@
+            } if (value === SentryConsentState.Refused) {
+                console.log('[SentryClient] Synchronised consent state is Refused. Shutting down the client');
+                FbwAircraftSentryClient.closeSentry();
+            }
+        });
+
+        return Promise.resolve(false);
+    }
+
+    /**
+     * Runs the root client flow, checking if consent state is known, asking for it otherwise, then initializing appropriately
+     *
+     * @param config a {@link FbwAircraftSentryClientConfiguration} object
+     *
+     * @returns a `Promise<boolean>` that indicates the result of the root client flow
+     */
+    private async runRootClientFlow(config: FbwAircraftSentryClientConfiguration): Promise<boolean> {
+        const consentValue = NXDataStore.get(SENTRY_CONSENT_KEY, SentryConsentState.Unknown) as SentryConsentState;
+
+        switch (consentValue) {
+        case SentryConsentState.Given:
+            console.log('[SentryClient] Consent state is Given. Initializing sentry');
+
+            return FbwAircraftSentryClient.attemptInitializeSentry(config);
+        case SentryConsentState.Unknown:
+            console.log('[SentryClient] Consent state is Unknown. Asking for consent');
+
+            return FbwAircraftSentryClient.requestConsent().then((didConsent) => {
+                if (didConsent) {
+                    NXDataStore.set(SENTRY_CONSENT_KEY, SentryConsentState.Given);
+
+                    console.log('[SentryClient] User requested consent state Given. Initializing sentry');
+
+                    return FbwAircraftSentryClient.attemptInitializeSentry(config);
+                }
+
+                NXDataStore.set(SENTRY_CONSENT_KEY, SentryConsentState.Refused);
+
+                console.log('[SentryClient] User requested consent state Refused. Doing nothing');
+
+                return false;
+            }).catch(() => false);
+        case SentryConsentState.Refused:
+            console.log('[SentryClient] Consent state is Refused. Doing nothing');
+            break;
+        default:
+            console.log('[SentryClient] Consent state is corrupted. Doing nothing');
+            break;
+        }
+
+        return false;
+    }
+
+    /**
+     * Displays an MSFS popup request consent for error reporting
+     *
+     * @returns a `Promise<boolean` indicating the consent state
+     */
+    static async requestConsent() {
+        const popup = new PopUp();
+
+        return new Promise<boolean>((resolve) => {
+            popup.showPopUp(
+                'A32NX - ERROR REPORTING',
+                'Are you willing to help us by enabling anonymous error reporting? This is 100% optional and we will never collect your personal data, '
+                + 'but it will help us diagnose issues quickly.',
+                'normal',
+                () => resolve(true),
+                () => resolve(false),
+            );
+        });
+    }
```

Comment:
Can you clarify to the user what you want them to do by asking them a question? Currently they just see a Yes and No button Yes what? No what?

Reply:
Sorry was looking at the old message